inc/Test/Base.pm

```diff
@@ -438,14 +438,32 @@ sub _block_list_init {
 
 sub _choose_blocks {
     my $blocks = [];
+    my $has_first = 0;
+
+    for my $hunk (@_) {
+        my $block = $self->_make_block($hunk);
+        if (exists $block->{FIRST}) {
+            $has_first = 1;
+            last;
+        }
+    }
+
     for my $hunk (@_) {
         my $block = $self->_make_block($hunk);
         if (exists $block->{ONLY}) {
             diag "I found ONLY: maybe you're debugging?"
                 unless $self->_no_diag_on_only;
             return [$block];
         }
+
         next if exists $block->{SKIP};
+        if ($has_first) {
+            if (exists $block->{FIRST}) {
+                $has_first = 0;
+            }
+            next;
```

Comment:
```suggestion
            } else {
                next;
            }
```
Okay, I got you @zhuizhuhaomeng 
Do you mean this change?

Reply:
Yes

Reply:
Yes, that's what I mean